tools/java/org/unicode/cldr/util/StandardCodes.java

```diff
@@ -1073,6 +1075,17 @@ boolean isWellFormed(String candidate) {
         public String toCompatString() {
             return this == region ? "territory" : toString();
         }
+
+        public static LstrType fromString(String rawType) {
+            try {
+                return valueOf(rawType);
+            } catch (Exception e) {
```

Comment:
Do the if check first and just don't have a try-catch block.
valueOf() is defined to only throw IllegalArgumentException or NullPointerException and I think you want to complain if you're given null anyway (it's a very different type of failure mode).